server/models/Board.js

```diff
@@ -43,6 +43,37 @@ BoardSchema.statics.deleteBoard = function (boardId) {
   this.deleteOne({ _id: mongoose.Types.ObjectId(boardId) }).exec();
 };
 
+// gets a Board by _id
+BoardSchema.statics.findBoard = async function (boardId) {
+  const foundBoard = await this.findById(boardId);
```

Comment:
```suggestion
  const foundBoard = await this.findById(boardId).populate({ 
     path: 'columns',
     populate: {
       path: 'cards',
       model: 'Card'
     } 
  });
```
This is a much easier way to accomplish this, but clever implementation!